### PR TITLE
Cleanup Crafting Stations Internals

### PR DESCRIPTION
**Pull request by Paneedah**
## 📝 Description

Begins to clean up the code related to crafting and the workbench.

## 🎯 Goals

Make the code more readable and optimize the code to make it somewhat more efficient.

The following **Base Files** for stations:
- [x] BlockStation
- [x] GUIContainerStation
- [x] TileEntityStation
- [x] CraftingStation (/mwc/groovyscript/script/CraftingStation.java)

The following **Workbench Specific Files**:
- [x] ContainerWorkbench
- [x] CustomSearchTextField
- [x] GUIButtonCustom
- [x] GUIContainerWorkbench
- [x] PagesSlot
- [x] PagedSlotItemHandler
- [x] TESRWorkbench
- [x] WorkbenchBlock

Bigger things:
- [x] Fix Luna's port of workbench (can craft when button is disabled).
- [x] Clean up WorkbenchServerMessageHandler
- [x] Clean up WorkbenchClientMessageHandler

## ❌ Non Goals

Adding new features, editing the UI of the workbenches, or any of the front-end.

## 🚦 Testing 

Ran the mod after editing every file and followed the following:

1. Try craft gun with empty workbench.
2. Try craft gun with all items but insufficient.
3. Try craft gun with one item but sufficient quantity.
4. Try craft gun with sufficient items (all within one stack).
5. Try craft gun with sufficient items (all scattered between stacks).
6. Mixture of 4 and 5.
7. Try craft more than 1 at a time.

## ⏮️ Backwards Compatibility 

Since this is a workbench PR, this will be perfectly backwards compatible.
There is nothing here that *can* cause issues with updating. No weapons have been added/removed.

## 📚 Related Issues & Documents

N/A

## 📖 Added to documentation?

- [ ] 📜 README.md
- [ ] 📑 Documentation
- [ ] 📓 Javadoc
- [ ] 🍕 Comments
- [x] 🙅 No documentation needed

## Summary by CodeRabbit

- **New Features**
	- Updated the positioning of GUI elements to enhance user interface experience.
	- Improved the search functionality within the crafting interface for better usability.
	- Enhanced crafting information tooltips for clarity and detail.

- **Bug Fixes**
	- Added null check to prevent potential crashes during block harvesting.
	- Fixed typo in variable naming for material availability checks.
	- Corrected text input handling in the search field to improve filtering of crafting list results.

- **Refactor**
	- Improved readability and maintainability of various methods across the crafting system.
	- Shifted towards immutability in class fields to ensure stability and predictability.
	- Restructured internal logic for crafting timer and progress tracking for better performance.

- **Style**
	- Conducted minor formatting adjustments for a more consistent codebase appearance.
	- Removed redundant code comments and unnecessary line breaks to clean up the code.

- **Documentation**
	- Updated error messages and comments to better guide users through crafting processes.

- **Chores**
	- Renamed methods and variables to adhere to naming conventions and improve code clarity.

### CODE REVIEW

**Comment by Paneedah** [comment]
Testing - Please do not review just yet.

File: src/main/java/com/paneedah/weaponlib/crafting/workbench/TileEntityWorkbench.java [hunk 1]
```diff
@@ -11,32 +11,24 @@
...
 	public String craftingTargetName;
 
-
 	public int ticker;
 
-
-
-	public TileEntityWorkbench() {
-	}
+	public TileEntityWorkbench() {}
```
[the remark this answers is not in the document]
Comment (Paneedah) [reply src/main/java/com/paneedah/weaponlib/crafting/workbench/TileEntityWorkbench.java:21]: Thanks coderabbit! I forgor to remove it.

File: src/main/java/com/paneedah/mwc/groovyscript/script/CraftingStation.java [hunk 2]
```diff
@@ -91,12 +91,12 @@ public static class RecipeBuilder extends AbstractRecipeBuilder<GSCrafting> {
 
         @Override
         public String getErrorMsg() {
-            return "Error adding Modern Warefare Cubed Workbench Recipe: "+this.name;
+            return "Error adding Modern Warfare Cubed Workbench Recipe: " + this.name;
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/groovyscript/script/CraftingStation.java:94]: Displaying the name of the mod should be done using the constant in `ProjectConstants`

File: src/main/java/com/paneedah/mwc/network/handlers/WorkbenchServerMessageHandler.java [hunk 3]
```diff
@@ -54,7 +53,6 @@ public IMessage onMessage(final WorkbenchServerMessage workbenchServerMessage, f
                             press.addStack(newStack);
 
                         CHANNEL.sendToAllAround(new WorkbenchClientMessage(station.getWorld(), workbenchServerMessage.getTeLocation()), new TargetPoint(0, workbenchServerMessage.getTeLocation().getX(), workbenchServerMessage.getTeLocation().getY(), workbenchServerMessage.getTeLocation().getZ(), 20));
-
                         return;
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/network/handlers/WorkbenchServerMessageHandler.java:56]: Nah, this spacing was good

File: src/main/java/com/paneedah/mwc/network/handlers/WorkbenchServerMessageHandler.java [hunk 4]
```diff
@@ -66,10 +64,11 @@ public IMessage onMessage(final WorkbenchServerMessage workbenchServerMessage, f
 
                     // Calculate the itemstacks to remove
                     for (CraftingEntry stack : modernRecipe) {
-                        itemRemovalList.computeIfAbsent(stack.getItem(), k -> new HashMap<>());
                         final Item stackItem = stack.getItem();
                         final int requiredCount = stack.getCount();
 
+                        itemRemovalList.computeIfAbsent(stackItem, k -> new HashMap<>());
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/network/handlers/WorkbenchServerMessageHandler.java:70]: No don't switch that around

File: src/main/java/com/paneedah/mwc/network/handlers/WorkbenchServerMessageHandler.java [hunk 5]
```diff
@@ -111,8 +110,8 @@ public IMessage onMessage(final WorkbenchServerMessage workbenchServerMessage, f
                     }
 
                     station.sendUpdate();
-
                     CHANNEL.sendToAllAround(new WorkbenchClientMessage(station.getWorld(), workbenchServerMessage.getTeLocation()), new TargetPoint(0, workbenchServerMessage.getTeLocation().getX(), workbenchServerMessage.getTeLocation().getY(), workbenchServerMessage.getTeLocation().getZ(), 20));
+
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/network/handlers/WorkbenchServerMessageHandler.java:114]: Those spaces changes are not good

File: src/main/java/com/paneedah/mwc/network/handlers/WorkbenchServerMessageHandler.java [hunk 6]
```diff
@@ -126,13 +125,15 @@ public IMessage onMessage(final WorkbenchServerMessage workbenchServerMessage, f
                     }
 
                     CHANNEL.sendToAllAround(new WorkbenchClientMessage(station.getWorld(), workbenchServerMessage.getTeLocation()), new TargetPoint(0, workbenchServerMessage.getTeLocation().getX(), workbenchServerMessage.getTeLocation().getY(), workbenchServerMessage.getTeLocation().getZ(), 25));
+
                 } else if (workbenchServerMessage.getOpCode() == WorkbenchServerMessage.MOVE_OUTPUT) {
                     ((EntityPlayer) world.getEntityByID(workbenchServerMessage.getPlayerID())).addItemStackToInventory(station.mainInventory.getStackInSlot(workbenchServerMessage.getSlotToMove()));
+
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/network/handlers/WorkbenchServerMessageHandler.java:131]: Don't add spaces here

File: src/main/java/com/paneedah/weaponlib/crafting/base/BlockStation.java [hunk 7]
```diff
@@ -47,43 +45,35 @@ public BlockStation(ModContext context, String name, Material materialIn) {
...
+			enumfacing = EnumFacing.SOUTH;
+
+		else if (enumfacing == EnumFacing.SOUTH && iblockstate1.isFullBlock() && !iblockstate.isFullBlock())
+			enumfacing = EnumFacing.NORTH;
+
+		else if (enumfacing == EnumFacing.WEST && iblockstate2.isFullBlock() && !iblockstate3.isFullBlock())
+			enumfacing = EnumFacing.EAST;
+
+		else if (enumfacing == EnumFacing.EAST && iblockstate3.isFullBlock() && !iblockstate2.isFullBlock())
+			enumfacing = EnumFacing.WEST;
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/weaponlib/crafting/base/BlockStation.java:72]: Use brackets for this, and don't add spaces, under

File: src/main/java/com/paneedah/weaponlib/crafting/base/GUIContainerStation.java [hunk 8]
```diff
@@ -116,43 +115,41 @@ public void initGui() {
...
+		this.craftButton.setErrored(true);
 
-		addButton(craftButton);
-		addButton(leftArrow);
-		addButton(rightArrow);
-		addButton(dismantleButton);
+		addButton(this.craftButton);
+		addButton(this.leftArrow);
+		addButton(this.rightArrow);
+		addButton(this.dismantleButton);
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/weaponlib/crafting/base/GUIContainerStation.java:147]: Use `this` only when necessary

File: src/main/java/com/paneedah/weaponlib/crafting/base/GUIContainerStation.java [hunk 9]
```diff
@@ -206,56 +203,58 @@ public boolean hasSelectedCraftingPiece() {
...
-
 		}
 
-		if(requiresMaterialsToSubmitCraftRequest()) this.craftButton.setErrored(!hasRequiredItems);
-		else this.craftButton.setErrored(false);
-
+		if (requiresMaterialsToSubmitCraftRequest())
+      		this.craftButton.setErrored(!hasRequiredItems);
+		else 
+        	this.craftButton.setErrored(false);
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/weaponlib/crafting/base/GUIContainerStation.java:257]: Use brackets, and correct spacing

**Review by Desoroxxx — CHANGES_REQUESTED** [review]

File: src/main/java/com/paneedah/weaponlib/crafting/workbench/ContainerWorkbench.java [hunk 10]
```diff
@@ -44,52 +43,47 @@ public ContainerWorkbench(EntityPlayer player, InventoryPlayer inventory, TileEn
...
+        return ItemStack.EMPTY;
 	}
 
+	//	Panda: Redundant - Same as superclass.
+	//
+	//	@Override
+	//	public boolean canInteractWith(EntityPlayer playerIn) {
+	//		return true;
+	//	}
+	//
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/weaponlib/crafting/workbench/ContainerWorkbench.java:88]: Remove this

**Review by Desoroxxx — CHANGES_REQUESTED** [review]

**Pull request merged**